reVX/wind_setbacks/wind_setbacks.py

```diff
@@ -0,0 +1,1386 @@
+        """
+        logger.debug('Generating setbacks exclusion array of shape {}'
+                     .format(self._shape))
+        arr = features.rasterize(shapes=shapes,
+                                 out_shape=self._shape[1:],
+                                 fill=0,
+                                 transform=self._profile['transform'],
+                                 dtype='uint8')
+
+        return np.expand_dims(arr, axis=0)
+
+    def _general_setbacks(self, features, max_workers=None):
+        """
+        Compute general setbacks for all states either in serial or parallel
+
+        Parameters
+        ----------
+        features : list
+            list of feature files to compute setbacks from
+        max_workers : int, optional
+            Number of workers to use for setback computation, if 1 run in
+            serial, if > 1 run in parallel with that many workers, if None
+            run in parallel on all available cores, by default None
+
+        Returns
+        -------
+        setbacks : ndarray
+            Raster array of setbacks, ready to be written to the exclusions
+            .h5 file as a new exclusion layer
+        """
+        if max_workers is None:
+            max_workers = os.cpu_count()
+
+        crs = self._profile['crs']
+
+        setbacks = []
+        if max_workers > 1:
+            logger.info('Computing general setbacks in parallel using {} '
+                        'workers'.format(max_workers))
+            loggers = [__name__, 'reVX']
+            with SpawnProcessPool(max_workers=max_workers,
+                                  loggers=loggers) as exe:
+                futures = []
+                for state_features in features:
+                    future = exe.submit(self._compute_general_setbacks,
+                                        state_features, crs,
+                                        self.generic_setback)
+                    futures.append(future)
+
+                for i, future in enumerate(as_completed(futures)):
+                    setbacks.extend(future.result())
+                    logger.debug('Computed setbacks for {} of {} states'
+                                 .format((i + 1), len(features)))
+        else:
+            logger.info('Computing general setbacks in serial')
+            for i, state_features in enumerate(features):
+                setbacks.extend(self._compute_general_setbacks(
+                    state_features, crs, self.generic_setback))
+                logger.debug('Computed setbacks for {} of {} states'
+                             .format((i + 1), len(features)))
+
+        setbacks = self._rasterize_setbacks(setbacks)
+
+        return setbacks
+
+    def _existing_setbacks(self, features_state_map, groupby='State',
+                           max_workers=None):
+        """
+        Compute existing setbacks for all states either in serial or parallel
```

Comment:
What does this do? The exclusions with setbacks are already in a shape file and you're just rasterizing them? Digging into the code i think this is calculating setbacks from pre-existing structures with pre-existing state setback regulations, but that's not how this reads to me.